src/CoinbaseSmartWalletFactory.sol

```diff
@@ -46,12 +42,13 @@ contract CoinbaseSmartWalletFactory {
         }
 
         (bool alreadyDeployed, address accountAddress) =
-            LibClone.createDeterministicERC1967(msg.value, implementation, _getSalt(owners, nonce));
+            LibClone.createDeterministicERC1967(msg.value, emptyImplementation, _getSalt(owners, nonce, implementation));
```

Comment:
It's important for security that the original implementation address is in the hash. Otherwise anyone could deploy an account to your address with whatever implementation they want. 

Currently this means that a users address across many chains always has to start at the same version in order to have the same address. And so the UX gain is mainly that new users can start at a certain version without needing to upgrade to it.

Reply:
Actually the whole EmptyUUPS is moot if we're doing this. May as well just pass `implementation` here